CMakeLists.txt

```diff
@@ -159,10 +159,26 @@ elseif(HTTPLIB_USE_BROTLI_IF_AVAILABLE)
 endif()
 
 if(HTTPLIB_REQUIRE_ZSTD)
-	find_package(zstd REQUIRED)
+	find_package(zstd)
+	if(NOT zstd_FOUND)
+		find_package(PkgConfig REQUIRED)
+		pkg_check_modules(zstd REQUIRED IMPORTED_TARGET libzstd)
+		add_library(zstd::libzstd ALIAS PkgConfig::zstd)
+	endif()
```

Comment:
This PR appears to be "obviously broken", because it is obvious that attempting to do `find_package(cpp-httplib)` doesn't include the same changes, thus, it fails.

Aside: maybe about time for the cmake files to generate a pkgconfig file?

Reply:
The zstd project doesn't distribute a find_package() module at all, so the default case involves building with a pkgconf dependency.

Reply:
@eli-schwartz that may add support of usage via `pkg_check_modules(cpp-httplib)` or however it would be called, but wouldn't fix `find_package(httplib)` unless it's rewritten to use pkg-config, if I understand what you mean

Reply:
It would also allow usage from autoconf or meson. ;)

Meson can execute `find_package(httplib)` as a dependency finder but it is quite slow and fragile as it involves generating a dummy project which does so, and then using the cmake debug traces to determine what compile arguments were resolved by `find_package(httplib)`. Using a buildsystem-agnostic interchange format allows divorcing oneself from the unreliability of a cmake config file...

Reply:
[All build systems suck](https://xkcd.com/927), it's not really worth talking about.
Even this project has like 3 I think? Cmake, Meson, and Makefile.
If you want a `.pc` file then write one, I doubt the PR would be denied.

Also, as I said, Cmake doesn't make generating them any less painless then the current `cmake.in` config system. It'd have the same problems (and probably more janky).

Reply:
So erm maybe if all build systems suck that's a reason to support pkg-config (which is just a standard, not a set of competing standards)???

Because, you know. Then I can find cpp-httplib and use it in my code ***without*** having to deal with buildsystem proliferation, since I'm not required to run a build system in order to find the compile flags for cpp-httplib. I can even use:

```
gcc my-code.cpp -o my-program $(pkg-config --cflags --libs cpp-httplib)
```

"Look ma, no build system!"